src/net_processing.cpp

```diff
@@ -2007,10 +2008,19 @@ void static ProcessOrphanTx(CConnman* connman, CTxMemPool& mempool, std::set<uin
             // Has inputs but not accepted to mempool
             // Probably non-standard or insufficient fee
             LogPrint(BCLog::MEMPOOL, "   removed orphan tx %s\n", orphanHash.ToString());
-            if (orphanTx.HasWitness() || orphan_state.GetResult() != TxValidationResult::TX_WITNESS_MUTATED) {
-                // Do not use rejection cache for witness transactions or
-                // witness-stripped transactions, as they can have been malleated.
-                // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
+            if (orphan_state.GetResult() != TxValidationResult::TX_WITNESS_STRIPPED) {
+                // Do not add txids of witness transactions or witness-stripped
```

Comment:
Isn't this block about tracking `witness txid` not `txid`?

Reply:
Ok I guess recentRejects is pretty confusing.  Made one more attempt to add comments around how this works.

Reply:
I understand how they work(in master), I didn't understand why it was talking about something it's not doing. With the added comment it's clearer, thanks.